File: ProjetFinal/src/main/java/projetFinal/entities/CommandeADomicile.java
package projetFinal.entities;

import java.time.LocalDate;
import java.util.Set;

public class CommandeADomicile extends Reservation{
	private Adresse adresse;
	private Set<ItemMenu> itemsMenu;
	
	public CommandeADomicile() {
		super();
	}

	public CommandeADomicile(Client client, Restaurant restaurant, LocalDate date, String specification,
			Adresse adresse, Set<ItemMenu> itemsMenu) {
		super(client, restaurant, date, specification);
		this.adresse = adresse;
		this.itemsMenu = itemsMenu;
	}
	
	public CommandeADomicile(Long id_reservation, Client client, Restaurant restaurant, LocalDate date, String specification,
			Adresse adresse, Set<ItemMenu> itemsMenu) {
		super(id_reservation,client, restaurant, date, specification);
		this.adresse = adresse;
		this.itemsMenu = itemsMenu;
	}

	public Adresse getAdresse() {
		return adresse;
	}

	public void setAdresse(Adresse adresse) {
		this.adresse = adresse;
	}

	public Set<ItemMenu> getItemsMenu() {
		return itemsMenu;
	}

	public void setItemsMenu(Set<ItemMenu> itemsMenu) {
		this.itemsMenu = itemsMenu;
	}
}

File: projetFinalAngular/src/app/model/commande-adomicile.ts
import { Adresse } from './adresse';
import { HeureReservation } from './heure-reservation';
import { ItemMenu } from './item-menu';
import { Reservation } from './reservation';
import { Restaurant } from './restaurant';
import { Utilisateur } from './utilisateur';

export class CommandeADomicile extends Reservation {
  public get itemsMenu(): ItemMenu[] | undefined {
    return this._itemsMenu;
  }
  public set itemsMenu(value: ItemMenu[] | undefined) {
    this._itemsMenu = value;
  }
  public get adresse(): Adresse {
    return this._adresse;
  }
  public set adresse(value: Adresse) {
    this._adresse = value;
  }

  constructor(
    private _adresse: Adresse,
    _client: Utilisateur,
    _id?: number,
    _restaurant?: Restaurant,
    _date?: Date,
    _specification?: string,
    private _itemsMenu?: ItemMenu[]
  ) {
    super(_client, _id, _restaurant, _date, _specification);
  }
}
